src/main/scala/com/mozilla/telemetry/heka/package.scala

```diff
@@ -7,13 +7,95 @@
 package com.mozilla.telemetry
 
 import com.google.protobuf.ByteString
+import org.json4s._
+import org.json4s.jackson.JsonMethods._
+import org.json4s.JsonDSL._
+import scala.util.{Try, Success, Failure}
 
 package object heka {
   class RichMessage(m: Message) {
+
+    /** Deserialize the Heka message as a native Scala Map. */
     def fieldsAsMap: Map[String, Any] = {
       val fields = m.fields
       Map(fields.map(_.name).zip(fields.map(field)): _*)
     }
+
+    /** Reconstructs a telemetry Heka message as a json4s JValue.
+      *
+      * The message is parsed and serialized into JSON. Keys that have been extracted
+      * into the toplevel document are inserted into to the proper place in the document.
+      *
+      * @return wrapped reconstructed JValue
+      */
+    def toJValue: Try[JValue] = {
+      val fields = m.fields.map(f => (f.name, fieldAsJValue(f))).toMap
+
+      // Parse the document payload. All pings using v4 of the telemetry infrastructure will use
+      // the "submission" field instead of the message payload. This kept for backwards compatibility
+      // with tests and older telemetry data (est. cut-over is February 2017).
+      val payload = m.payload match {
+        case Some(payload: String) => Try(parse(payload))
+        case _ => {
+          fields.get("submission") match {
+            case Some(submission) => submission match {
+              // submission field must be a json object if it exists
+              case JObject(_) => Success(submission)
+              case _ => Failure(new MappingException("submission field is not an object"))
```

Comment:
You should either throw the exception without wrapping it or make this value None